entropylab/tests/test_complex_db.py

```diff
@@ -0,0 +1,60 @@
+from entropylab import Script, Graph
+from entropylab.api.data_reader import ExperimentReader
+from entropylab.api.execution import EntropyContext
+from entropylab.graph_experiment import PyNode
+from entropylab.instruments.lab_topology import ExperimentResources, LabResources
+from entropylab.results_backend.sqlalchemy.db import SqlAlchemyDB
+import numpy as np 
+import subprocess
+
+db_file=r'C:\repos\entropy\entropylab\tests\myproj'
+db=SqlAlchemyDB(path=db_file)
+er = ExperimentResources(db)    
+
+## 1d array, single node
+def node_operation():
+    return {'res':np.array([1,2,3,4])}
+
+node1 = PyNode(label="first_node", program=node_operation,output_vars={'res'})
+experiment = Graph(resources=er, graph={node1}, story="run_a") 
+handle = experiment.run()
+
+
+
+## 2d array, single node
+def node_operation():
+    return {'res':np.random.randn(5,5)}
+
+node1 = PyNode(label="2d array", program=node_operation,output_vars={'res'})
+experiment = Graph(resources=er, graph={node1}, story="run_a") 
+handle = experiment.run()
+
+
+## 3d array, single node
+def node_operation():
+    return {'res':np.random.randn(5,5,3)}
+
+node1 = PyNode(label="3d array", program=node_operation,output_vars={'res'})
+experiment = Graph(resources=er, graph={node1}, story="run_a") 
+handle = experiment.run()
+
+## 2 1d arrays, 1 2d array , single node
+def node_operation():
+    return {'x':np.random.randn(10),'y':np.random.randn(10),'res3':np.random.randn(10,10)}
+
+node1 = PyNode(label="multiple res", program=node_operation,output_vars={'res'})
+experiment = Graph(resources=er, graph={node1}, story="run_a") 
+handle = experiment.run()
+
+
+##2 nodes
+def node_operation1():
+    return {'x':np.random.randn(10),'y':np.random.randn(10),'res3':np.random.randn(10,10)}
+
+def node_operation2():
+    return {'x':np.random.randn(10),'y':np.random.randn(10),'res3':np.random.randn(10,10)}
+
+node1 = PyNode("the first node", node_operation1, output_vars={"x"})
+node2 = PyNode("the second node", node_operation2, input_vars={"x": node1.outputs["x"]})
+experiment = Graph(resources=er, graph={node1,node2}, story="multi-node") 
+graph_handle = experiment.run(db)
```

Comment:
What will we be asserting each experiment run? Let's formalize it into an `assert` statement.